PhpAmqpLib/Connection/AbstractConnection.php

```diff
@@ -180,11 +180,15 @@ public function __construct(
         $login_method = 'AMQPLAIN',
         $login_response = null,
         $locale = 'en_US',
-        AbstractIO $io,
+        AbstractIO $io = null,
```

Comment:
Why is this needed?

Reply:
see https://github.com/php-amqplib/php-amqplib/issues/846#issuecomment-752987677 for example, there are at least 3 other duplicate reports of the same issue.

Reply:
Sorry, I misexplained my thought.

This argument is already required (and not nullable) so all previous arguments must be provided.

It seems that removing the default value for all previous arguments could avoid the `is_null` check without breaking compatibility.

Reply:
Basically I agree with you. It is not needed.
The single reason for this change is to make this PR minor-version-mergable.